Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,233 @@
+
+        self.logger.log_info("Shutdown complete")
+
+    def get_sampling_params_dict(self, params_json):
+        """
+        This functions parses the dictionary values into their
+        expected format.
+        """
+
+        dict = json.loads(params_json)
+
+        # Special parsing for the supported sampling parameters
+        # TODO: Add more parameters if needed
+        dict["temperature"] = float(dict["temperature"])
+        dict["top_p"] = float(dict["top_p"])
+
+        return dict
+
+    def postprocess(self, request_output):
+        """
+        Parses the output from the vLLM engine into text
+        response.
+        """
+        prompt = request_output.prompt
+        text_outputs = [
+            (prompt + output.text).encode("utf-8") for output in request_output.outputs
+        ]
+        return text_outputs
+
+    async def generate(self, request):
+        """
+        Forwards single request to LLM engine and returns responses.
+        """
+        response_sender = request.get_response_sender()
+        self.ongoing_request_count += 1
+        try:
+            request_id = random_uuid()
+            prompt = pb_utils.get_input_tensor_by_name(request, "PROMPT").as_numpy()[0]
+            stream = pb_utils.get_input_tensor_by_name(request, "STREAM").as_numpy()[0]
+            sampling_params_dict = self.get_sampling_params_dict(request.parameters())
+            sampling_params = SamplingParams(**sampling_params_dict)
+            results_generator = self.llm_engine.generate(
+                str(prompt), sampling_params, request_id
+            )
+
+            # Streaming case
+            async def stream_results() -> AsyncGenerator[bytes, None]:
+                async for request_output in results_generator:
+                    if self._shutdown_event.is_set():
+                        await self.llm_engine.abort(request_id)
+                        yield None
+                    else:
+                        output = self.postprocess(request_output)
+                        triton_output_tensor = pb_utils.Tensor(
+                            "TEXT", np.asarray(output, dtype=self.output_dtype)
+                        )
+                        yield pb_utils.InferenceResponse(
+                            output_tensors=[triton_output_tensor]
+                        )
+
+            if stream:
+                async for response in stream_results():
+                    if response:
+                        response_sender.send(response)
+            # Non-Streaming case
+            else:
+                final_output = None
+                async for request_output in results_generator:
+                    if self._shutdown_event.is_set():
+                        await self.llm_engine.abort(request_id)
+                    else:
+                        final_output = request_output
+                if final_output:
+                    output = self.postprocess(final_output)
```

Comment:
this block of code is redundant with the similar block above.

best to combine the logic into a single function to avoid needing to maintain it in two places.